include/mbedtls/ssl.h

```diff
@@ -1672,6 +1687,13 @@ struct mbedtls_ssl_context
     /** Callback to export key block and master secret                      */
     mbedtls_ssl_export_keys_t *MBEDTLS_PRIVATE(f_export_keys);
     void *MBEDTLS_PRIVATE(p_export_keys);            /*!< context for key export callback    */
+
+    /** User data pointer or handle.
+     *
+     * The library sets this to \p 0 when creating a context and does not
+     * access it afterwards.
+     */
+    uintptr_t user_data;
```

Comment:
If literally **every** other member of `mbedtls_ssl_context` is MBEDTLS_PRIVATE, I would recommend creating getter/setter accessors for the new member `user_data` so that the entire struct/object remains opaque.  Okay to have inline funcs in ssl.h to access these members, but at least code that migrates will be code compatible.  If you want to hide or changes these in 4.0, then calling code just needs to be recompiled, not modified.